### PR TITLE
Geos area reduction

### PR DESCRIPTION
**Pull request by mraspaud**
Implement area-slicing for geos-based areas.

 - [x] Tests added 
 - [x] Tests passed 
 - [x] Passes ``git diff origin/develop **/*py | flake8 --diff`` 
 - [x] Fully documented

### CODE REVIEW

**Comment by mraspaud** [comment]
I wrote this a long time ago, but iirc it was because I wanted to go directly from the instrument scanning angle to the lon/lat coordinates. But you're right, we could use proj and just divide the x and y projection coordinates by the altitude of the satellite to get the angle.

About `crop_around` vs `crop`, my idea was that just `crop` would be equivalent to `__getitem__`, and the `around` felt more intuitive name-wise when providing an area

**Comment by davidh-ssec** [comment]
I was thinking of not using satellite height at all. We should be able to use transform (https://jswhit.github.io/pyproj/pyproj-module.html#transform). This would only require the source PROJ.4, the target PROJ.4, and the extents of the two areas. We could convert from the crop area extents to the source area extents and use those to figure out the indexes of the area to crop. If any of the crop area extents can't be transformed (1e30 iirc) then we use the source area extent value for that coordinate.

I thought maybe `crop` could accept multiple keyword arguments `crop(self, area=None, lonlat_bbox=None, xy_bbox=None):` and then `getitem` would be for row/col index numbers.

**Comment by davidh-ssec** [comment]
Summary of conversation with @mraspaud on slack: The boundary objects already handle everything that would need to be handled if we used the X/Y PROJ.4 transforms. They also use spherical coordinates (I think) so it should work in all cases.

File: pyresample/geometry.py [hunk 1]
```diff
@@ -1369,6 +1370,107 @@ def proj4_string(self):
...
+                         ' identical: %s', area_to_cover.proj_dict['proj'])
+            # Get xy coordinates
+            llx, lly, urx, ury = area_to_cover.area_extent
+            x, y = self.get_xy_from_proj_coords([llx, urx], [lly, ury])
+
+            return slice(x[0], x[1] + 1), slice(y[1], y[0] + 1)
+
+        from trollsched.boundary import AreaDefBoundary, Boundary
+
+        data_boundary = Boundary(*get_geostationary_bounding_box(self))
```
[the remark this answers is not in the document]
Comment (djhoese) [reply pyresample/geometry.py]: Additionally, if the ROI is `geos` then shouldn't that also use the above logic `Boundary(*get_geostation_bounding_box(...))`?

**Pull request merged**